src/NServiceBus.Core/Transports/Msmq/MessagePump.cs

```diff
@@ -146,18 +146,16 @@ public void Start(PushRuntimeSettings limitations)
 
                     await concurrencyLimiter.WaitAsync(cancellationToken).ConfigureAwait(false);
 
+                    var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationTokenSource.Token);
```

Comment:
That is a good point @johnsimons , how about we apply SRP and only use this one to allow the pipeline to request the receive to abort?

This means:
1. It should not be a linked token
2. If we ever support the transport to "signal" that it's shutting down it would be a different token

Thoughts?

@Scooletz @Particular/nservicebus